crates/core/src/table/mod.rs

```diff
@@ -520,6 +520,72 @@ impl Table {
             })
     }
 
+    /// Get all the changed [FileSlice]s in splits from the table between the given timestamps.
+    ///
+    /// # Arguments
+    ///     * `n` - The number of chunks to split the file slices into.
+    ///     * `start_timestamp` - If provided, only file slices that were changed after this timestamp will be returned.
+    ///     * `end_timestamp` - If provided, only file slices that were changed before or at this timestamp will be returned.
```

Comment:
Is it the same semantics (start_timestamp exclusive, end_timestamp inclusive) across other engines? Start inclusive, end exclusive feels more intuitive but if our semantics match across engines, I am fine with it.